File: PDOO/NapakalakiRuby.ruby/lib/death_bad_consequence.rb
#encoding: utf-8

require_relative 'numeric_bad_consequence'

module NapakalakiGame
  
  class DeathBadConsequence < NumericBadConsequence
    
    def initialize(b)
      bc=super(b, Player.getMAXLEVEL, MAXTREASURES, MAXTREASURES)
    end
    
    def getDeath
      true
    end
        
    def to_s
      cadena = super 
      cadena += "\nDeath: #{@death}"
end
end
end

File: PDOO/Napakalaki.java/src/NapakalakiGame/DeathBadConsequence.java
package NapakalakiGame;

/**
 *
 * @author Andres Arco Lopez
 */
public class DeathBadConsequence extends NumericBadConsequence{
    
    public DeathBadConsequence(String t){
        super(t,Player.MAXLEVEL, MAXTREASURES, MAXTREASURES);
    }
    
    @Override
    public boolean getDeath(){
        return true;
    }
    
    @Override
    public String toString(){
        String mensaje = super.toString();
        mensaje += "\nMuerto: " + getDeath();
        
        return mensaje;
    }
    
}
